### PR TITLE
Feature/return file cx 678

### PR DESCRIPTION
**Pull request by stefaniacardenas**
I've just changed the branch name, now the build is green.

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/index.js [hunk 1]
```diff
@@ -103,23 +103,31 @@ class Capture extends Component {
     this.validateCapture(message.id, valid)
   }
 
-  handleImage = (base64ImageLossy, base64capture, file) => {
-    if (!base64capture) {
+  handleBase64 = (lossyBase64, base64) => {
+    base64toFile(base64,
+      file => this.handleImage(lossyBase64, file)
```
Comment (hjheath) [inline src/components/Capture/index.js]: I still find this hard to follow. Can you give the callback function a name?

e.g. `const fileCallback = file => this.handleImage(lossyBase64, file)`

**Comment by hjheath** [comment]
Cool looks good. Was it simple to upload the result of the SDK to the API? Can you update the changelog?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/index.js (hunk 1 above)
Comment (stefaniacardenas) [inline src/components/Capture/index.js]: Or I could remove the callback and do 
```
const file = base64toFile
this.handleImage(lossyBase64, file)

```

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/index.js (hunk 1 above)
Comment (stefaniacardenas) [inline src/components/Capture/index.js]: Actually, I think that passing the anonymous function is best practice and much cleaner

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Confirm/index.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ const Capture = ({capture:{image, file}}) =>
   <div className={style.captures}>
     {file && isOfFileType(['pdf'], file) ?
```
Comment (rfreitas) [inline src/components/Confirm/index.js]: this no longer works, since file does not exist anymore

File: src/components/utils/file.js [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
-  const reader = new FileReader();
-  reader.readAsDataURL(file);
-  reader.onload = function () {
-    callback(reader.result)
-  };
-  reader.onerror = function (error) {
-    console.log('File Reading Error: ', error);
-    errorCallback(error)
-  };
+const decodeBase64 = (image) => {
```
Comment (rfreitas) [inline src/components/utils/file.js]: it might be worth profiling this for large images. If slow, consider slicing the processing into multiple cycles and turning this function into something async.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Capture/index.js [hunk 4]
```diff
@@ -148,31 +153,34 @@ class Capture extends Component {
...
-          error => handleImageBound(undefined)
-        )
-      }
-      else {
-        handleImageBound(undefined)
-      }
+    if (isOfFileType(['pdf'], file)){
+      // TODO: we still need to convert PDFs  to base64 in order to send the lossyBase64 to the websocket server
+      // this code needs to be changed when HTTP protocol is implemented
+      const handlePDF = (lossyBase64) => this.handleImage(lossyBase64, file)
```
Comment (rfreitas) [inline src/components/Capture/index.js]: `lossyBase64` probably not the right name, just `base64File`

File: src/components/Capture/index.js [hunk 5]
```diff
@@ -103,22 +103,27 @@ class Capture extends Component {
...
-    const payload = this.createPayload(base64capture, base64ImageLossy, file)
+    const payload = this.createPayload(image, base64ImageLossy)
     functionalSwitch(this.props.method, {
       document: ()=> this.handleDocument(payload),
       face: ()=> this.handleFace(payload)
     })
   }
 
-  createPayload = (image, imageLossy, file) => ({
+  createPayload = (image, imageLossy) => ({
```
Comment (rfreitas) [inline src/components/Capture/index.js]: consider renaming `image` to `file`

File: src/components/utils/file.js [hunk 6]
```diff
@@ -2,15 +2,34 @@ import loadImage from 'blueimp-load-image/js/load-image'
...
-  const reader = new FileReader();
-  reader.readAsDataURL(file);
-  reader.onload = function () {
-    callback(reader.result)
-  };
-  reader.onerror = function (error) {
-    console.log('File Reading Error: ', error);
-    errorCallback(error)
-  };
+   const reader = new FileReader();
```
Comment (hjheath) [inline src/components/utils/file.js]: There's no actual diff here right? Is the indentation off?

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/utils/file.js (hunk 6 above)
Comment (stefaniacardenas) [inline src/components/utils/file.js]: Yes, there was an indentation issue. Fixed

File: src/components/Capture/index.js (hunk 5 above)
Comment (stefaniacardenas) [inline src/components/Capture/index.js]: Done

File: src/components/Capture/index.js (hunk 4 above)
Comment (stefaniacardenas) [inline src/components/Capture/index.js]: Done

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/utils/file.js [hunk 7]
```diff
@@ -5,12 +5,31 @@ export const fileToBase64 = (file, callback, errorCallback) => {
...
+  return {integerArray, mimeString}
+}
+
+export const base64toFile = (image, callback) => {
+  const base64Data = decodeBase64(image)
+  const blob = new Blob([base64Data.integerArray], {type: base64Data.mimeString})
+  const file = new File([blob], 'image.'+ blob.type.split('/')[1], {type: base64Data.mimeString})
+  const preview = URL.createObjectURL(blob)
+  file.preview = preview
+  callback(file)
```
Comment (rfreitas) [inline src/components/utils/file.js]: would not use a callback if the code is not async

Comment (rfreitas) [inline src/components/utils/file.js]: would not use a callback if the code is not async

**Comment by rfreitas** [comment]
Apart from the minor "fake" async function, it looks pretty good.

File: src/components/utils/file.js (hunk 7 above)
Comment (stefaniacardenas) [inline src/components/utils/file.js]: Got rid of the fake async callback ;)

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: CHANGELOG.md [hunk 8]
```diff
@@ -6,6 +6,10 @@ Breaking changes result in a different major. UI changes that might break custom
 
 ## [next-version]
 
+## Changed
+
+- Public: Document and face captures will be returned by callbacks as File objects instead of base64.
+
```
Comment (rfreitas) [inline CHANGELOG.md]: a selector was changed: `.image` to `.file`

Comment (stefaniacardenas) [inline CHANGELOG.md]: I've changed the class name back to .image for backward compatibility

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://135-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/utils/file.js [hunk 9]
```diff
@@ -5,12 +5,31 @@ export const fileToBase64 = (file, callback, errorCallback) => {
...
+  for (let i = 0; i < byteString.length; i++) {
+    integerArray[i] = byteString.charCodeAt(i)
+  }
+  return {integerArray, mimeString}
+}
+
+export const base64toFile = (image) => {
+  const base64Data = decodeBase64(image)
+  const blob = new Blob([base64Data.integerArray], {type: base64Data.mimeString})
+  const file = new File([blob], 'image.'+ blob.type.split('/')[1], {type: base64Data.mimeString})
```
Comment (rfreitas) [inline src/components/utils/file.js]: @stephencookdev it seems it will break on IE 11...

Comment (rfreitas) [inline src/components/utils/file.js]: @stefaniacardenas this will break on IE 11
